envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -62,25 +62,30 @@ end
 
 function envoy_on_response(handle)
     local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac") or {}
+    local is_shadow_denied = (rbacMetadata["shadow_engine_result"] or "") == "denied"
     local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
     local allowed_client = lua_metadata["request.info.allowed_client"] or false
+    local upstream_request_time = handle:headers():get("x-envoy-upstream-service-time")
+    local rbac_action = "shadow_denied"
```

Comment:
`rbac_action` code can be moved inside `if should_log_request()`, so it will be not unnecessary evaluated for all requests.